src/frontend/data/sideEffects/upload/upload.spec.ts

```diff
@@ -0,0 +1,286 @@
+import fetchMock from 'fetch-mock';
+import xhrMock from 'xhr-mock';
+
+import { modelName } from '../../../types/models';
+import { timedTextMode, uploadState, Video } from '../../../types/tracks';
+import { upload } from './upload';
+
+describe('upload', () => {
+  const object = {
+    description: 'Some description',
+    id: 'video-id',
+    is_ready_to_play: true,
+    show_download: false,
+    thumbnail: null,
+    timed_text_tracks: [
+      {
+        active_stamp: 1549385921,
+        id: 'ttt-1',
+        is_ready_to_play: true,
+        language: 'fr',
+        mode: timedTextMode.SUBTITLE,
+        upload_state: uploadState.READY,
+        url: 'https://example.com/timedtext/ttt-1.vtt',
+      },
+      {
+        active_stamp: 1549385922,
+        id: 'ttt-2',
+        is_ready_to_play: false,
+        language: 'fr',
+        mode: timedTextMode.SUBTITLE,
+        upload_state: uploadState.READY,
+        url: 'https://example.com/timedtext/ttt-2.vtt',
+      },
+      {
+        active_stamp: 1549385923,
+        id: 'ttt-3',
+        is_ready_to_play: true,
+        language: 'en',
+        mode: timedTextMode.CLOSED_CAPTIONING,
+        upload_state: uploadState.READY,
+        url: 'https://example.com/timedtext/ttt-3.vtt',
+      },
+      {
+        active_stamp: 1549385924,
+        id: 'ttt-4',
+        is_ready_to_play: true,
+        language: 'fr',
+        mode: timedTextMode.TRANSCRIPT,
+        upload_state: uploadState.READY,
+        url: 'https://example.com/timedtext/ttt-4.vtt',
+      },
+    ],
+    title: 'Some title',
+    upload_state: 'pending',
+    urls: {
+      manifests: {
+        dash: 'https://example.com/dash.mpd',
+        hls: 'https://example.com/hls.m3u8',
+      },
+      mp4: {
+        144: 'https://example.com/144p.mp4',
+        1080: 'https://example.com/1080p.mp4',
+      },
+      thumbnails: {
+        720: 'https://example.com/144p.jpg',
+      },
+    },
+  } as Video;
+
+  beforeEach(jest.resetAllMocks);
+  beforeEach(() => xhrMock.setup());
+
+  afterEach(fetchMock.restore);
+  afterEach(() => xhrMock.teardown());
+  it('gets the policy from the API and uses it to upload the file', async () => {
+    const updateObject = jest.fn();
+    const setStatus = jest.fn();
+    const notifyObjectUploadProgress = jest.fn();
+    const jwt = 'foo';
+    const objectType = modelName.VIDEOS;
+
+    const file = new File(['(⌐□_□)'], 'course.mp4', { type: 'video/mp4' });
+
+    fetchMock.mock(
+      `/api/${modelName.VIDEOS}/${object.id}/initiate-upload/`,
```

Comment:
IMO it's better to hardcode the values in the tests instead of replicating the code in the actual files